tests/unit/api/main.py

```diff
@@ -132,6 +132,54 @@ def test_duplicate_project_create(self):
         with pytest.raises(api.DuplicateError):
             api.project_create('acme-corp')
 
+class TestProjectAddDeleteNetwork:
```

Comment:
We should be moving away from this style of test, where we populate the objects needed by the test *in* the test; it makes it harder to see at a glance what the test is actually looking for. This is why I created `initial_db`. If you need additional objects to write these tests, you can add them to `additional_db`, but please don't add more tests like this. They were a mistake when we wrote them, let's not make it worse.

I wrote a blog post inspired by this file a while back:

https://zenhack.net/2016/03/26/test-code-is-still-code.html